tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -106,11 +106,14 @@ def enhance_policies(self, access_policies):
         # GraphHelper.get_principal_dictionary returns empty AADObject if not found with graph
         # or if graph is not available.
         principal_dics = GraphHelper.get_principal_dictionary(self.graph_client, object_ids)
-
+        remove = []
         for policy in access_policies:
             aad_object = principal_dics[policy['objectId']]
+            if aad_object.object_id is None:
```

Comment:
I would not remove the access policy and fill properties with 'N/A' or something similar to make sure it doesn't just go through the filter.

It feels wrong to ignore item with some access level to the KeyVault.

Reply:
I've modified it to now return false if the access policies can't be found

Reply:
I think we are solving 2 potential issues here right?
1. We don't have access to the graph - This probably should just fail the policy
2. The principal doesn't exist - In which case, we are writing out a warning and then removing the access policy

Reply:
To Kirill's point, we shouldn't remove the policy because the behavior of the the filter is...

"We don't care what permissions the user has if it is part of the list of users; if the user is not part of that list, at most it should the the following permissions."

If you remove the unfound_principals (these are the ones returned from KeyVault query), then we will not do the second check of the behavior above. Which means, these access_policies could have permissions that violate the whitelist.

Reply:
i agree with failing the policy in the case of ambiguity via exception raise, in general that should be a default behavior. Ideally the two cases would be distinguishable, although if the context is api principal not existing, that should have been handled earlier in execution.  In the case of a principal as part of a data set (not api call) with non existing the error should be avoided via an additional data set filtering check. in the case of a resource based permission the case is bit more nuanced,  if its common then a log is okay, . There's some ongoing discussion on formalizing and extending those semantics in another open issue (aka data collection on error and resource).